src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -123,6 +125,31 @@ public string HostId
             }
         }
 
+        /// <summary>
+        /// Gets the host machine ID. All host instances on the same machine will have the same HostMachineId.
+        /// </summary>
+        public string HostMachineId
+        {
+            get
+            {
+                if (_hostMachineId == null)
+                {
+                    _hostMachineId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")
+                        ?? Environment.MachineName.GetHashCode().ToString("X", CultureInfo.InvariantCulture).PadLeft(32, '0');
```

Comment:
It must be a Guid to be used as a proposed lease id. 

From the docs (https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/lease-blob) : x-ms-proposed-lease-id: Proposed lease ID, in a GUID string format.